aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1182 @@
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*4);
+        vest(contract_address);
+        vested_amount_1 = vested_amount_1 + fraction(shareholder_1_share, 1, 10);
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*5);
+        vest(contract_address);
+        vested_amount_1 = vested_amount_1 + fraction(shareholder_1_share, 1, 10);
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*6);
+        vest(contract_address);
+        vested_amount_1 = vested_amount_1 + fraction(shareholder_1_share, 1, 10);
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*7);
+        vest(contract_address);
+        vested_amount_1 = vested_amount_1 + fraction(shareholder_1_share, 1, 10);
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*8);
+        vest(contract_address);
+        vested_amount_1 = vested_amount_1 + fraction(shareholder_1_share, 1, 10);
+        vested_amount_2 = vested_amount_2 + fraction(shareholder_2_share, 1, 10);
+        assert!(coin::balance<AptosCoin>(shareholder_1_address) == vested_amount_1, 0);
+        assert!(coin::balance<AptosCoin>(shareholder_2_address) == vested_amount_2, 0);
+
+        timestamp::update_global_time_for_test_secs(vesting_start_secs(contract_address)+period_duration_secs(contract_address)*9);
+        vest(contract_address);
```

Comment:
I am just wondering why the vest function is called 10 times here. I understand that the first call is made before the time of vesting so we can ignore it. However, there are still 9 vest calls remaining, but it should be 8 call because **vesting_numerators** is `[2, 2, 1]` and **vesting_denominator** is `10`.

If I remove the last vest call, the process fails. @axiongsupra, could you please provide further clarification on this.

Reply:
@nizam-supraoracles This is because we do calculations on fixed_point32 to get the vested amount, which is not precise enough.

Please see each step of two vested_amout after the call to vest.

```
[debug] 0
[debug] 0
[debug] 49
[debug] 149
[debug] 98
[debug] 298
[debug] 122
[debug] 372
[debug] 146
[debug] 446
[debug] 170
[debug] 520
[debug] 194
[debug] 594
[debug] 218
[debug] 668
[debug] 242
[debug] 742
[debug] 250
[debug] 750
```
You can also try add `debug::print(&vested_amount_1);` to your code to see.

Reply:
Okay @axiongsupra, fair enough. I thought there should be a vested_amount as below
```
[debug] 0
[debug] 0
[debug] 50
[debug] 150
[debug] 100
[debug] 300
[debug] 125
[debug] 375
[debug] 150
[debug] 450
[debug] 175
[debug] 525
[debug] 200
[debug] 600
[debug] 225
[debug] 675
[debug] 250
[debug] 750
```
 However, as you mentioned, we do calculations on `fixed_point32` to determine the vested amount, which is not precise enough.

FYI, Dr. @sjoshisupra Dr. @davidsupra